js/shared_controls.js

```diff
@@ -553,7 +589,10 @@ function getZMoveName(moveName, moveType, item) {
 														moveName === "Thunderbolt" && item === "Aloraichium Z" ? "Stoked Sparksurfer" :
 															moveName === "Thunderbolt" && item === "Pikashunium Z" ? "10,000,000 Volt Thunderbolt" :
 																moveName === "Volt Tackle" && item === "Pikanium Z" ? "Catastropika" :
-																	ZMOVES_TYPING[moveType];
+																	moveName === "Nature\'s Madness" && item === "Tapunium Z" ? "Guardian of Alola" :
+																		moveName === "Nature\'s Madness" && item !== "Tapunium Z" ? "Nature\'s Madness" :
+																			ZMOVES_TYPING[moveType];
+																			// What if we had Nature's Madness and the user checked the Z move option? The calc will be tricked into Twinkle Tackle.
```

Comment:
um

Reply:
Nvm ^^".

Reply:
It is obsolete now, see the next commit.